### ISSUE TITLE
[sec] Decide on which Gunicorn settings to process and expose

### ISSUE THREAD

**Issue by sbellem** [bigchaindb/bigchaindb#1379]
**tl;dr**: BigchainDB's HTTP server can be configured in insecure ways. For example it could be vulnerable to DDOS attacks. The root cause lies in the fact that all Gunicorn settings can currently be set in arbitrary ways via BigchainDB's current configuration mechanism. 

---
At the time of writing we have:

```python
# in processes.py
def start():
    # ...
    # start the web api
    app_server = server.create_server(bigchaindb.config['server'])
    p_webapi = mp.Process(name='webapi', target=app_server.run)
    p_webapi.start()
    # ...
```

the code of the `create_server` function contains:

```python
def create_app(*, debug=False, threads=4):
    # ...
    standalone = StandaloneApplication(app, settings)
    return standalone
```

finally, the code of the `StandaloneApplication` class is such that at init time, the `load_config()` method will be invoked via the `gunicorn` parent class `BaseApplication`'s method [`do_load_config()`](https://github.com/benoitc/gunicorn/blob/master/gunicorn/app/base.py#L17-L) and the code of `load_config()` is such that it will set whatever gunicorn settings is found in the given `options`:

```python
    def load_config(self):
        config = dict((key, value) for key, value in self.options.items()
                      if key in self.cfg.settings and value is not None)

        for key, value in config.items():
            # not sure if we need the `key.lower` here, will just keep
            # keep it for now.
            self.cfg.set(key.lower(), value)
```

this probably has both advantages and disadvantages. One key advantage is that someone could configure Gunicorn as much as Gunicorn allows it. One obvious disadvantage is that someone could potentially configure Gunicorn in a way that is detrimental to the overall BigchainDB system. A more precise example would be the security setting [`limit_request_field_size`](http://docs.gunicorn.org/en/latest/settings.html#limit-request-field-size) for which Gunicorn itself gives a warning:

> Setting this parameter to a very high or unlimited value can open up for DDOS attacks.

Hence this issue wishes to address the question:

**"Which Gunicorn settings can be configured?"**

This is the "write" part so to speak. As for the "read" part, we may wish to expose both configurable and non-configurable settings, hence the question:

**"Which Gunicorn settings can be viewed?"**

If we were to decide that certain Gunicorn settings cannot be configured via the BigchainDB configuration mechanism then we will need to implement a filter in the configuration mechanism that will only consider those settings that are "allowed".
